src/plone/folder/default.py

```diff
@@ -34,8 +34,11 @@ def notifyRemoved(self, obj_id):
         """ see interfaces.py """
         order = self._order()
         pos = self._pos()
-        idx = pos[obj_id]
-        del order[idx]
+        try:
+            idx = pos[obj_id]
+            del order[idx]
+        except KeyError:
+            pass
```

Comment:
I do not like idx stays undefined if the exception is raised. 
`del order[pos[obj_id]]` or  `order.pop(pos[obj_id], None)` would avoid the need of the temp idx variable.
Anyway not a blocker.